client/src/App.tsx

```diff
@@ -1,18 +1,57 @@
+import { useEffect, useState } from 'react';
 import { Routes, Route } from 'react-router-dom';
-import './App.css';
+import AppContext from './components/AppContext';
 import NavBar from './components/NavBar';
 import JobDetails from './pages/JobDetails';
 import Jobs from './pages/Jobs';
+import AuthPage from './pages/AuthPage';
+import './App.css';
+import { Auth, User } from './lib';
+import Home from './pages/Home';
+
+const tokenKey = 'designengineer';
 
 export default function App() {
+  const [user, setUser] = useState<User>();
+  const [token, setToken] = useState<string>();
+  const [isAuthorizing, setIsAuthorizing] = useState(true);
+
+  useEffect(() => {
```

Comment:
Since localStorage is synchronous, you might get away with not having a useEffect. But it doesn't hurt.